lib/array_intersection.rb

```diff
@@ -1,4 +1,24 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  intersection = []
 
-end
\ No newline at end of file
+  if list1.length < list2.length
+    smaller_list = list1
+    larger_list = list2
+  else
+    larger_list = list1
+    smaller_list = list2
+  end
+
+  lookup = {}
+  smaller_list.each do |number|
+    lookup[number] = ""
+  end
+
+  larger_list.each do |number|
+    if lookup.include?(number)
+      intersection << number
+    end
```

Comment:
Instead of using `.include?` I would just do this for readability and because `.include?` makes people think it's doing a linear search. 

```suggestion
    unless lookup[number].nil?
      intersection << number
    end
```